src/utils/openmc/material.py

```diff
@@ -0,0 +1,491 @@
+            raise ValueError(msg)
+
+        if not percent_type in ['ao', 'wo']:
+            msg = 'Unable to add an Element to Material ID={0} with a ' \
+                  'percent type {1}'.format(self._id, percent_type)
+            raise ValueError(msg)
+
+        # Copy this Element to separate it from same Element in other Materials
+        element = deepcopy(element)
+
+        self._elements[element._name] = (element, percent, percent_type)
+
+
+    def remove_element(self, element):
+
+        # If the Material contains the Element, delete it
+        if element._name in self._elements:
+            del self._elements[element._name]
+
+
+    def add_s_alpha_beta(self, name, xs):
+
+        if not is_string(name):
+            msg = 'Unable to add an S(a,b) table to Material ID={0} with a ' \
+                        'non-string table name {1}'.format(self._id, name)
+            raise ValueError(msg)
+
+        if not is_string(xs):
+            msg = 'Unable to add an S(a,b) table to Material ID={0} with a ' \
+                  'non-string cross-section identifier {1}'.format(self._id, xs)
+            raise ValueError(msg)
+
+        self._sab.append((name, xs))
+
+
+    def set_otf_mat_file(self, name):
+
+        # TODO: remove this when distributed materials are merged
+        warnings.warn('This feature is not yet implemented in a release ' \
+                      'version of openmc')
+
+        if not is_string(name) and not name is None:
+            msg = 'Unable to add OTF material file to Material ID={0} with a ' \
+                        'non-string name {1}'.format(self._id, name)
+            raise ValueError(msg)
+
+        self._distrib_otf_file = name
+
+
+    def set_as_distrib_comp(self):
+
+        # TODO: remove this when distributed materials are merged
+        warnings.warn('This feature is not yet implemented in a release ' \
+                      'version of openmc')
+
+        self._convert_to_distrib_comps = True
+
+
+    def get_all_nuclides(self):
+
+        nuclides = dict()
+
+        for nuclide_name, nuclide_tuple in self._nuclides.items():
+            nuclide = nuclide_tuple[0]
+            density = nuclide_tuple[1]
+            nuclides[nuclide._name] = (nuclide, density)
+
+        return nuclides
+
+
+    def _repr__(self):
+
+        string = 'Material\n'
+        string += '{0: <16}{1}{2}\n'.format('\tID', '=\t', self._id)
+        string += '{0: <16}{1}{2}\n'.format('\tName', '=\t', self._name)
+
+        string += '{0: <16}{1}{2}'.format('\tDensity', '=\t', self._density)
+        string += ' [{0}]\n'.format(self._density_units)
+
+        string += '{0: <16}'.format('\tS(a,b) Tables') + '\n'
+
+        for sab in self._sab:
+            string += '{0: <16}{1}[{2}{3}]\n'.format('\tS(a,b)', '=\t',
+                                                     sab[0], sab[1])
+
+        string += '{0: <16}'.format('\tNuclides') + '\n'
```

Comment:
Also it is weird to mix tabs and whitespace padding. Are literal tabs actually needed?